File: Java/Medium/WrongAnswer/maximum-binary-string-after-change_3.java
class Solution {
    public String maximumBinaryString(String binary) {
        
        int n = binary.length();
        int count = 0;
        int first = 0;
        for (int i = 0; i < n; i++) {
            if (binary.charAt(i) == '0') {
                count++;
                if (first == 0) {
                    first = i;
                }
            }
        }
        if (count == 0) {
            return binary;
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < first; i++) {
            sb.append('1');
        }
        sb.append('0');
        for (int i = 0; i < count - 1; i++) {
            sb.append('1');
        }
        for (int i = first + count; i < n; i++) {
            sb.append('0');
        }
        return sb.toString();
    }
}

File: C++/Medium/WrongAnswer/maximum-binary-string-after-change_3.cpp
class Solution {
public:
    string maximumBinaryString(string binary) {
        
        int n = binary.size();
        int count = 0;
        int i = 0;
        while (i < n && binary[i] == '1') {
            i++;
        }
        while (i < n) {
            if (binary[i] == '0') {
                count++;
            }
            i++;
        }
        string res = "";
        for (int j = 0; j < i - count - 1; j++) {
            res += "1";
        }
        res += "0";
        for (int j = 0; j < count; j++) {
            res += "1";
        }
        return res;
    }
};